docker/gaffer-gremlin/build.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+# Copyright 2023 Crown Copyright
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+declare SCRIPT_DIR="$(readlink -f "$(dirname "${0}")")"
```

Comment:
Not sure how this Dockerfile differs from the one that already exists? Also seems to rely on running a script rather than doing the build within this file, which is what the other Dockerfiles do

Reply:
It farms out the Jar download to maven to make the dockerfile way simpler as it can then just copy them in, this script just ties them together neatly with some error checking. The other dockerfiles do a build with a clone of the git repo which is unnecessary as the Jars are already built and in maven central.

Reply:
The other Dockerfiles only do a local build if fetching from central failed. This is to facilitate development by allowing the version to be set to `develop` or `my-branch`.

Reply:
You can still do a development build using this method too so long as the snapshot build is at least in your local maven cache e.g. via `install` which if you're doing development on the main repo I would expect it would be checked out locally already. This does partly beg the question would it not be better to just have the docker stuff in the main repo? Most projects don't separate them out like this.

Reply:
We have discussed it before but it is quite a lot of content to put into the main repo, especially considering all of the docker compose examples and helm scripts.

Reply:
That's fair I would say its not a massive about of files in the grand scheme of things but probably a wider discussion outside of this PR. However, I do still feel using maven to collect the dependencies is a more appropriate method than the existing hardcoded url to fat jar and/or git clone method, as it provides a clear boundary between the main code and here, it also means develop changes don't have to be pushed to the main repo in order to do development builds as well.